datasets/code_x_glue_cc_clone_detection_big_clone_bench/code_x_glue_cc_clone_detection_big_clone_bench.py

```diff
@@ -0,0 +1,90 @@
+from typing import List
+
+import datasets
+
+from .common import TrainValidTestChild
+from .generated_definitions import DEFINITIONS
+
+
+class CodeXGlueCCCloneDetectionBigCloneBench(TrainValidTestChild):
+    _DESCRIPTION = """Given two codes as the input, the task is to do binary classification (0/1), where 1 stands for semantic equivalence and 0 for others. Models are evaluated by F1 score.
+The dataset we use is BigCloneBench and filtered following the paper Detecting Code Clones with Graph Neural Network and Flow-Augmented Abstract Syntax Tree."""
+
+    _CITATION = """@inproceedings{svajlenko2014towards,
+  title={Towards a big data curated benchmark of inter-project code clones},
+  author={Svajlenko, Jeffrey and Islam, Judith F and Keivanloo, Iman and Roy, Chanchal K and Mia, Mohammad Mamun},
+  booktitle={2014 IEEE International Conference on Software Maintenance and Evolution},
+  pages={476--480},
+  year={2014},
+  organization={IEEE}
+}
+
+@inproceedings{wang2020detecting,
+  title={Detecting Code Clones with Graph Neural Network and Flow-Augmented Abstract Syntax Tree},
+  author={Wang, Wenhan and Li, Ge and Ma, Bo and Xia, Xin and Jin, Zhi},
+  booktitle={2020 IEEE 27th International Conference on Software Analysis, Evolution and Reengineering (SANER)},
+  pages={261--271},
+  year={2020},
+  organization={IEEE}
+}"""
+    _FEATURES = {
+        "id": datasets.Value("int32"),  # Index of the sample
+        "id1": datasets.Value("int32"),  # The first function id
+        "id2": datasets.Value("int32"),  # The second function id
+        "func1": datasets.Value("string"),  # The full text of the first function
+        "func2": datasets.Value("string"),  # The full text of the second function
+        "label": datasets.Value("bool"),  # 1 is the functions are not equivalent, 0 otherwise
+    }
+
+    _SUPERVISED_KEYS = ["label"]
+
+    def generate_urls(self, split_name):
+        yield "index", f"{split_name}.txt"
+        yield "data", "data.jsonl"
+
+    def _generate_examples(self, split_name, file_pathes):
+        import json
+
+        js_all = {}
+
+        with open(file_pathes["data"]) as f:
+            for idx, line in enumerate(f):
+                entry = json.loads(line)
+                js_all[int(entry["idx"])] = entry["func"]
+
+        with open(file_pathes["index"]) as f:
+            for idx, line in enumerate(f):
+                line = line.strip()
+                idx1, idx2, label = [int(i) for i in line.split("\t")]
+                func1 = js_all[idx1]
+                func2 = js_all[idx2]
+
+                yield idx, dict(id=idx, id1=idx1, id2=idx2, func1=func1, func2=func2, label=(label == 1))
+
+
+CLASS_MAPPING = {
+    "CodeXGlueCCCloneDetectionBigCloneBench": CodeXGlueCCCloneDetectionBigCloneBench,
+}
+
+
+class CodeXGlueCCCloneDetectionBigCloneBenchMain(datasets.GeneratorBasedBuilder):
```

Comment:
The name of the class should be the camel case version of the dataset script name so

CodeXGlueCcCloneDetectionBigCloneBench

Also note that after doing changes in dataset class names or dataset script names, you have to regenerate the dataset_infos.json file.

Reply:
Yes, I have a single script that runs the whole thing, so no problem.
Maybe we should ensure this constraint in the dataset loading code, so it does not work if the class has not the right name ?

Reply:
We should indeed have something to check this constraint. It doesn't raise error but it's better to have names that match for debugging or to have consistent cache file names.